### ISSUE TITLE
Creating policies for specific bucket

### ISSUE THREAD

**Issue by iMajna** [Chaffelson/nipyapi#241]
* Nipyapi version: 0.15.0
* NiFi version: 1.11.4
* NiFi-Registry version:  0.6.0
* Python version: 3.6.8
* Operating System: Centos 7.6

### Description

I'm trying to create policies for newly created bucket which is connected to process group.
If bucket has no policies and neither one policy has even been assign to it code will fail with either:

- _No policy has be found for action_ if you try to get policies for that bucket
- _ValueError: Found multiple policies for '/buckets' with 'read'_ if you try to create new policy

### What I Did

**bucket_id**: b488d2fa-6f0c-4fd2-9d6c-911a012a7811

Executing this: 
```
nipyapi.security.get_access_policy_for_resource(
          resource='buckets',
          action='read', 
          r_id='b488d2fa-6f0c-4fd2-9d6c-911a012a7811', 
          service='registry', 
          auto_create=True)
```
Results in: 
`
HTTP response body: No policy found for action='read', resource='/buckets/b488d2fa-6f0c-4fd2-9d6c-911a012a7811'
`

Which is fine, because there aren't any policies. Since there are no policies you would like to create new policies so what you do:
```
nipyapi.security.create_access_policy(
         resource='buckets',
         action='read',
         r_id='b488d2fa-6f0c-4fd2-9d6c-911a012a7811', 
        service='registry')
```

and then this results in: 
`
ValueError: Found multiple policies for '/buckets' with 'read'.
`

So, from what I see, it searches and tries to create read policy in _/buckets_ instead of _/buckets/b488d2fa-6f0c-4fd2-9d6c-911a012a7811_. 

Based on documentation

> r_id (optional[str]) – if NiFi, the resource ID of the resource

it can really be that you cannot create policy for specific bucket and this can be _feature proposal_ then?

On the other side it also says: 

> r_id (Optional[str]) – The UUID of the resource, valid only if targeting NiFi resources

for **nipyapi.security.get_access_policy_for_resource** but check "Extra" below because it worked for "registry" as well 

**Extra**
When I try to manually create policy for bucket in Registry (through UI), immediately delete same policy and follow with execution of:
```
$ reg_policy_rw = nipyapi.security.get_access_policy_for_resource(
                                        resource='buckets',
                                        action='read',
                                        r_id='b488d2fa-6f0c-4fd2-9d6c-911a012a7811', 
                                        service='registry', 
                                        auto_create=True)

$ nipyapi.security.add_user_group_to_access_policy(
                    reg_group_rw_identity,
                    policy=reg_policy_rw,
                    service='registry'
                    )
```
Creation of policy for specific group would work, but well, my automation doesn't make any sense then :D 

### Urgency

Well, it's not urgent, but it would be nice to have to complete automation cycle.

**Comment by iMajna** [comment]
I think this is fixable with extending this: https://github.com/Chaffelson/nipyapi/blob/e13e1f35cdc313d139d687ee7feb30cf9af6490b/nipyapi/security.py#L698
with extending: 
this one 
`resource=r`
to this: 
`resource='/'.join([r, r_id]) if r_id else r`

**Comment by Chaffelson** [comment]
This makes good sense, if you want to go ahead and implement that would be great.
Sorry I hadn't seen this until now, not sure what has happened to my notifications lately.

**Comment by cgmckeever** [comment]
This is still open, yet the change seems to be present, however Im still getting the same error 

`create_access_policy`
```
ValueError: Found multiple policies for '/buckets' with 'read'.
```
`get_access_policy_for_resource("buckets"`
```
No policy found for action='read', resource='/buckets/a570c091-f05d-456e-95a
```


```
pip3 show nipyapi
Name: nipyapi
Version: 0.16.2
Summary: Nifi-Python-Api: A convenient Python wrapper for the Apache NiFi Rest API
Home-page: https://github.com/Chaffelson/nipyapi
Author: Daniel Chaffelson
Author-email: chaffelson@gmail.com
License: Apache Software License 2.0
Location: /usr/local/lib/python3.9/site-packages
Requires: lxml, requests, future, setuptools, packaging, ruamel.yaml, pysocks, docker, xmltodict, six
Required-by:
```

@iMajna

**Comment by cgmckeever** [comment]
I've found a NASTY workaround 

```
ap = nipyapi.security.create_access_policy("buckets/" + service_buckets[bucket_name].identifier,
                        "read",
                        r_id=None,
                        service=registry)
```

**Comment by iMajna** [comment]
@cgmckeever, its on my local branch 👎  

Give me a day to test it one more time although it is already working for us, but just whether I changed anything else then I'll make PR and hopefully it will be accepted and merged :)

**Comment by Chaffelson** [comment]
Returning to this old issue - can you please advise if this needs to be revisited

**Comment by cgmckeever** [comment]
wowo -- thanks for digging this back up. Two things
- Appears I had some sort of workaround (wasnt my issue to start)
- Im no longer the maintainer of this in practice -- so Im not sure
